ompi/mca/common/ompio/common_ompio_aggregators.c

```diff
@@ -1333,14 +1333,12 @@ int mca_common_ompio_prepare_to_group(ompio_file_t *fh,
        if (NULL == aggr_bytes_per_group_tmp) {
           opal_output (1, "OUT OF MEMORY\n");
           ret = OMPI_ERR_OUT_OF_RESOURCE;
-          free(end_offsets_tmp);
           goto exit;
        }
     decision_list_tmp = (int* )malloc (fh->f_init_num_aggrs * sizeof(int));
     if (NULL == decision_list_tmp) {
         opal_output (1, "OUT OF MEMORY\n");
         ret = OMPI_ERR_OUT_OF_RESOURCE;
-        free(end_offsets_tmp);
```

Comment:
This is not correct. THere is an explicit comment at the end of the file:
/* Do not free aggr_bytes_per_group_tmp,
** start_offsets_lens_tmp, and end_offsets_tmp
** here. The memory is released in the layer above.
*/
Did you run any tests to verify this? I just looked at the code, and I am pretty sure that the layer above it frees these arrays even in the case of an error correctly.

Reply:
These `free` calls were erroneous as they would have led to double frees. at this point, `end_offsets_tmp` has been assigned to the out parameter and would have been free'd again in the caller.